pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,443 @@
+// https://github.com/kubevirt/kubevirt/blob/main/vendor/github.com/spf13/pflag/flag.go#L983C1-L998C3
+func usage() string {
+	usage := "  # reset (to default) log-verbosity for all components\n"
+	usage += "  {{ProgramName}} adm logVerbosity --reset\n"
+
+	usage += "  # show log-verbosity for all components:\n"
+	usage += "  {{ProgramName}} adm log-verbosity --all\n"
+	usage += "  # set log-verbosity to 3 for all components:\n"
+	usage += "  {{ProgramName}} adm log-verbosity --all=3\n"
+
+	usage += "  # show log-verbosity for virt-handler:\n"
+	usage += "  {{ProgramName}} adm log-verbosity --virt-handler\n"
+	usage += "  # set log-verbosity to 7 for virt-handler:\n"
+	usage += "  {{ProgramName}} adm log-verbosity --virt-handler=7\n"
+
+	usage += "  # show log-verbosity for virt-handler and virt-launcher\n"
+	usage += "  {{ProgramName}} adm log-verbosity --virt-handler --virt-launcher\n"
+	usage += "  # set log-verbosity for virt-handler to 7 and virt-launcher to 3\n"
+	usage += "  {{ProgramName}} adm log-verbosity --virt-handler=7 --virt-launcher=3\n"
+
+	usage += "  # reset all components to default besides virt-handler which is 7\n"
+	usage += "  {{ProgramName}} adm log-verbosity --reset --virt-handler=7\n"
+	usage += "  # set all components to 3 besides virt-handler which is 7\n"
+	usage += "  {{ProgramName}} adm log-verbosity --all=3 --virt-handler=7\n"
+
+	return usage
+}
+
+// virtComponent to component name
+// also used by the test file
+func GetComponentNameByVirtComponent(component virtComponent) string {
+	var virtComponentToComponentName = map[virtComponent]string{
+		VirtAPI:        "virt-api",
+		VirtController: "virt-controller",
+		VirtHandler:    "virt-handler",
+		VirtLauncher:   "virt-launcher",
+		VirtOperator:   "virt-operator",
+		All:            "all",
+	}
+	return virtComponentToComponentName[component]
+}
+
+// virtComponent to JSON name
+func getJSONNameByVirtComponent(component virtComponent) string {
+	var virtComponentToJSONName = map[virtComponent]string{
+		VirtAPI:        "virtAPI",
+		VirtController: "virtController",
+		VirtHandler:    "virtHandler",
+		VirtLauncher:   "virtLauncher",
+		VirtOperator:   "virtOperator",
+		All:            "all",
+	}
+	return virtComponentToJSONName[component]
+}
+
+// component name to JSON name
+func getJSONNameByComponentName(componentName string) string {
+	var componentNameToJSONName = map[string]string{
+		"virt-api":        "virtAPI",
+		"virt-controller": "virtController",
+		"virt-handler":    "virtHandler",
+		"virt-launcher":   "virtLauncher",
+		"virt-operator":   "virtOperator",
+		"all":             "all",
+	}
+	return componentNameToJSONName[componentName]
+}
+
+func detectInstallNamespaceAndName(virtClient kubecli.KubevirtClient) (namespace, name string, err error) {
+	// listing KubeVirt CRs in all namespaces and see where it is installed
+	kvs, err := virtClient.KubeVirt("").List(&k8smetav1.ListOptions{})
```

Comment:
Is this listing in all namespaces or only in the default? Usually, empty strings are for the default namespace

Reply:
Thank you for finding this.

For some resources, the empty string is interpreted as “all namespaces”.
https://github.com/kubernetes/client-go/issues/159

And also, according to the following code, NamespaceAll is equal to “”.
https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L278

In some cases, it seems the empty string may be interpreted as “all namespaces”. However, I have not found a specific code to interpret the empty string in our case.

According to the page below, I think you are right. It should be treated as the “default namespace”.
https://kubevirt.io/api-reference/main/definitions.html#_k8s_io_apimachinery_pkg_apis_meta_v1_objectmeta

In any case, it's better to specify “all namespaces” explicitly. So, I will change the code as below to list KubeVirt CR across all namespaces.
`	kvs, err := virtClient.KubeVirt(k8smetav1.NamespaceAll).List(&k8smetav1.ListOptions{})`